src/common.cpp

```diff
@@ -217,5 +225,20 @@ instruction_ref add_common_op(module& m, const operation& op, std::vector<instru
     return insert_common_op(m, m.end(), op, std::move(inputs));
 }
 
+shape make_bcast_shape(const shape& input_shape,
+                       const std::vector<std::size_t>& bcast_lens,
+                       const std::size_t& offset)
+{
+    std::vector<size_t> bcast_strides(bcast_lens.size(), 0);
+    for(std::ptrdiff_t i = input_shape.ndim() - 1; i >= 0; i--)
```

Comment:
I think it would be nicer to write it as `for(std::ptrdiff_t i  : reverse(range(input_shape.ndim())))`.